application/common/components/SesMessage.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+
+namespace common\components;
+
+
+use yii\mail\BaseMessage;
+use yii\mail\MessageInterface;
+
+class SesMessage extends BaseMessage
+{
+    /** @var string */
+    private $_charset;
+
+    /** @var string */
+    private $_from;
+
+    /** @var string[] */
+    private $_to;
+
+    /** @var string[] */
+    private $_replyTo;
+
+    /** @var string[] */
+    private $_cc;
+
+    /** @var string[] */
+    private $_bcc;
+
+    /** @var string */
+    private $_subject;
+
+    /** @var string */
+    private $_textBody;
+
+    /** @var string */
+    private $_htmlBody;
+
+    public function getCharset()
+    {
+        return $this->_charset ?? 'UTF-8';
+    }
+
+    public function setCharset($charset)
+    {
+        $this->_charset = $charset;
+    }
+
+    public function getFrom()
+    {
+        return $this->_from;
+    }
+
+    public function setFrom($from)
+    {
+        if (is_array($from)) {
+            if (isset($from[0])) {
+                $this->_from = $from[0];
+            } else {
+                $addresses = array_keys($from);
+                $names = array_values($from);
+                $this->_from = sprintf('%s <%s>', $names[0], $addresses[0]);
+            }
+        } else {
+            $this->_from = $from;
+        }
+    }
+
+    public function getTo()
+    {
+        return $this->_to;
+    }
+
+    public function setTo($to)
+    {
+        if (is_array($to)) {
+            $this->_to = $to;
+        } else {
+            $this->_to = explode(",", $to);
+        }
+    }
+
+    public function getReplyTo()
+    {
+        return $this->_replyTo ?? [$this->getFrom()];
+    }
+
+    public function setReplyTo($replyTo)
+    {
+        if (is_array($replyTo)) {
+            $this->_replyTo = $replyTo;
+        } else {
+            $this->_replyTo = explode(",", $replyTo);
+        }
+    }
+
+    public function getCc()
```

Comment:
Include PHPDoc on things that return array, so you know the contents of the array.